File: CjERP.Backend/CjERP.Infrastructure/Services/PagoTesoreriaService.cs
using System.Data;
using CjERP.Application.DTOs;
using CjERP.Infrastructure.Persistence.Sql;
using Dapper;
using Microsoft.Data.SqlClient;

namespace CjERP.Infrastructure.Services;

public sealed partial class PagoTesoreriaService(ISqlCommandFactory factory)
{
    public async Task<object> CatalogosAsync(CancellationToken ct)
    {
        await using var cn = factory.CreateConnection();
        var ejecutores = await cn.QueryAsync(factory.Create("SELECT IdEmpleado AS Id, NombreEmpleado AS Nombre FROM Empleado WHERE IdCargo=14 AND IdEstado=1 ORDER BY NombreEmpleado", cancellationToken: ct));
        var constantes = (await cn.QueryAsync(factory.Create("SELECT Correlativo AS Id, ValorIni AS Nombre, TRY_CONVERT(decimal(18,4),ValorFin) AS Porcentaje, Campo FROM Constante WHERE Sociedad='PE01' AND Programa='PLANTILLA' AND Campo IN ('BANCO','TIPO_TRANSFERENCIA','TIPO_MONEDA','TIPO_COMPROBANTE','TIPO_PAGO','RENDICION','DETRACCION') AND Correlativo>=0 ORDER BY Correlativo", cancellationToken: ct))).ToList();
        return new { ejecutores, bancos = constantes.Where(x => x.Campo == "BANCO"), transferencias = constantes.Where(x => x.Campo == "TIPO_TRANSFERENCIA"), monedas = constantes.Where(x => x.Campo == "TIPO_MONEDA"), comprobantes = constantes.Where(x => x.Campo == "TIPO_COMPROBANTE"), tiposPago = constantes.Where(x => x.Campo == "TIPO_PAGO"), rendiciones = constantes.Where(x => x.Campo == "RENDICION"), retenciones = constantes.Where(x => string.Equals((string)x.Campo,"DETRACCION",StringComparison.OrdinalIgnoreCase)) };
    }

    public async Task<object> ListarAsync(int estado, DateTime? desde, DateTime? hasta, CancellationToken ct)
    {
        if (estado is not (1 or 9 or 8 or 5 or 4 or 2)) throw new ArgumentException("Estado de consulta inválido.");
        if (desde > hasta) throw new ArgumentException("La fecha inicial no puede superar la final.");
        if (estado == 4 && (desde is null || hasta is null)) throw new ArgumentException("Indique el rango de fechas del historial.");
        await using var cn = factory.CreateConnection();
        // Importes originales: la consulta general convierte dólares y agrupa operaciones bancarias.
        return await cn.QueryAsync(factory.Create($"""
            SELECT a.Correlativo, a.IdSite, a.Estado, a.TipoMoneda, a.IdResponsable,
                {VersionSql} AS Version,
                a.IdComprobante, a.IdTipoPago, a.Ruc, fechas.Emision AS FecEmision, a.IdRendicion, a.IdRetencion,
                a.IdBanco, a.IdTransferencia, a.IdMoneda2, a.RevisionPm, a.FechaRevision, a.Observacion, a.ImgFactura,
                a.RevisionPmAprobar, a.FechaRevisionAprobar, ren.ValorIni AS Rendicion,
                fechas.Ingreso AS Fecha, fechas.Deposito AS FechaDeposito, a.OT, a.Detalle, a.Serie,
                a.Subtotal, a.IGV, a.Total, a.MontoRetencion, a.TotalPagar,
                a.NroOperacion, a.Cheque, a.Comentario AS ComentarioAdicional,
                COALESCE(r.NombreEmpleado, a.Responsable) AS Responsable,
                CASE WHEN ISNULL(a.IdWeb,0)=1 THEN sc.NombreEmpleado ELSE s.NombreEmpleado END AS Solicitante,
                c.NombreCliente AS Cliente, p.NombreProyecto AS Proyecto, si.NombreSite AS Site,
                a.Tipo_Trabajo AS TipoTrabajo, tarea.ValorIni AS Tarea, doc.ValorIni AS Comprobante,
                mon.ValorIni AS Moneda, ban.ValorIni AS Banco, trans.ValorIni AS Transferencia
            FROM Planilla a
            CROSS APPLY (SELECT
                COALESCE(TRY_CONVERT(date,NULLIF(LTRIM(RTRIM(a.FecEmision)),''),23), TRY_CONVERT(date,NULLIF(LTRIM(RTRIM(a.FecEmision)),''),101), TRY_CONVERT(date,NULLIF(LTRIM(RTRIM(a.FecEmision)),''),103)) AS Emision,
                COALESCE(TRY_CONVERT(date,NULLIF(LTRIM(RTRIM(a.FecIngreso)),''),23), TRY_CONVERT(date,NULLIF(LTRIM(RTRIM(a.FecIngreso)),''),101), TRY_CONVERT(date,NULLIF(LTRIM(RTRIM(a.FecIngreso)),''),103)) AS Ingreso,
                COALESCE(TRY_CONVERT(date,NULLIF(LTRIM(RTRIM(a.FechaDeposito)),''),23), TRY_CONVERT(date,NULLIF(LTRIM(RTRIM(a.FechaDeposito)),''),101), TRY_CONVERT(date,NULLIF(LTRIM(RTRIM(a.FechaDeposito)),''),103)) AS Deposito
            ) fechas
            LEFT JOIN Empleado r ON r.IdEmpleado=a.IdResponsable
            LEFT JOIN Constante ren ON ren.Sociedad='PE01' AND ren.Programa='PLANTILLA' AND ren.Campo='RENDICION' AND ren.Correlativo=a.IdRendicion
            LEFT JOIN Empleado s ON s.IdEmpleado=a.IdSolicitante AND ISNULL(a.IdWeb,0)<>1
            LEFT JOIN EmpleadoCj sc ON sc.IdEmpleado=a.IdSolicitante AND ISNULL(a.IdWeb,0)=1
            LEFT JOIN Cliente c ON c.IdCliente=a.IdCliente
            LEFT JOIN Proyecto p ON p.IdProyecto=a.IdProyecto
            LEFT JOIN Site si ON si.IdSite=a.IdSite AND si.Correlativo=a.CorreSite
            LEFT JOIN Constante tarea ON tarea.Sociedad='PE01' AND tarea.Programa='PLANTILLA' AND tarea.Campo='TAREA' AND tarea.Correlativo=a.IdTarea
            LEFT JOIN Constante doc ON doc.Sociedad='PE01' AND doc.Programa='PLANTILLA' AND doc.Campo='TIPO_COMPROBANTE' AND doc.Correlativo=a.IdComprobante
            LEFT JOIN Constante mon ON mon.Sociedad='PE01' AND mon.Programa='PLANTILLA' AND mon.Campo='TIPO_MONEDA' AND mon.Correlativo=a.TipoMoneda
            LEFT JOIN Constante ban ON ban.Sociedad='PE01' AND ban.Programa='PLANTILLA' AND ban.Campo='BANCO' AND ban.Correlativo=a.IdBanco
            LEFT JOIN Constante trans ON trans.Sociedad='PE01' AND trans.Programa='PLANTILLA' AND trans.Campo='TIPO_TRANSFERENCIA' AND trans.Correlativo=a.IdTransferencia
            WHERE (a.Estado=@estado OR (@estado=2 AND a.Estado=7))
                AND (@desde IS NULL OR (CASE WHEN @estado=4 THEN fechas.Deposito ELSE fechas.Ingreso END)>=@desde)
                AND (@hasta IS NULL OR (CASE WHEN @estado=4 THEN fechas.Deposito ELSE fechas.Ingreso END)<=@hasta)
            ORDER BY a.Correlativo DESC
            """, new { estado, desde = desde?.Date, hasta = hasta?.Date }, cancellationToken: ct));
    }

    public async Task<object> CuentasAsync(int responsable, CancellationToken ct)
    {
        await using var cn = factory.CreateConnection();
        return await cn.QueryAsync(factory.Create("""
            SELECT DISTINCT c.Cuenta, c.CuentaInter, c.NombreCta, b.ValorIni AS Banco
            FROM CuentaEmpleado c
            LEFT JOIN Constante b ON b.Sociedad='PE01' AND b.Programa='PLANTILLA' AND b.Campo='BANCO' AND b.Correlativo=c.IdBanco
            WHERE c.IdEmpleado=@responsable AND c.Estado=1
            """, new { responsable }, cancellationToken: ct));
    }

    public async Task<int> PagarAsync(PagoTesoreriaRequestDto request, string usuario, CancellationToken ct)
    {
        if (request.EstadoOrigen is not (5 or 8)) throw new ArgumentException("Solo se pueden pagar recibos administrativos o programados.");
        if (request.Items.Count is < 1 or > 500 || request.Items.Select(x => x.Correlativo).Distinct().Count() != request.Items.Count)
            throw new ArgumentException("Seleccione entre 1 y 500 recibos sin duplicados.");
        if (request.FechaDeposito == default || request.FechaDeposito.Date > DateTime.UtcNow.AddHours(-5).Date)
            throw new ArgumentException("Indique una fecha de depósito válida, no futura.");
        await using var cn = factory.CreateConnection();
        await cn.OpenAsync(ct);
        await using var tx = (SqlTransaction)await cn.BeginTransactionAsync(ct);
        var ids = request.Items.Select(x => x.Correlativo).OrderBy(x => x).ToArray();
        var actuales = (await cn.QueryAsync<PagoActual>(new CommandDefinition($"""
            SELECT Correlativo, IdSite, Estado, TipoMoneda, TotalPagar, {VersionSql} AS Version
            FROM Planilla a WITH (UPDLOCK, HOLDLOCK) WHERE Correlativo IN @ids ORDER BY Correlativo
            """, new { ids }, tx, cancellationToken: ct))).ToList();
        if (actuales.Count != ids.Length || actuales.Any(a => a.Estado != request.EstadoOrigen || a.TipoMoneda != request.IdMoneda2 || a.TotalPagar <= 0 ||
            !request.Items.Any(i => i.Correlativo == a.Correlativo && i.IdSite.Trim() == a.IdSite.Trim() && i.TotalPagar == a.TotalPagar && i.Version == a.Version)))
            throw new InvalidOperationException("Los recibos cambiaron de estado, importe o moneda. Actualice la lista y revise la selección.");

        var comentariosExcedidos = await cn.ExecuteScalarAsync<int>(new CommandDefinition("""
            SELECT COUNT(*) FROM Planilla WHERE Correlativo IN @ids
            AND DATALENGTH(CONVERT(varchar(max),ISNULL(Comentario,'')) + ' - ' + CONVERT(varchar(max),@comentario)) > COL_LENGTH('dbo.Planilla','Comentario')
            """, new { ids, comentario = request.Comentario.Trim() }, tx, cancellationToken: ct));
        if (comentariosExcedidos > 0) throw new ArgumentException("El comentario supera el espacio disponible en uno de los recibos. Reduzca el texto y vuelva a revisar el lote.");

        var ejecutorValido = await cn.ExecuteScalarAsync<int>(new CommandDefinition("SELECT COUNT(*) FROM Empleado WHERE IdEmpleado=@IdEjecutor AND IdCargo=14 AND IdEstado=1", request, tx, cancellationToken: ct));
        var catalogosValidos = await cn.ExecuteScalarAsync<int>(new CommandDefinition("""
            SELECT COUNT(DISTINCT Campo) FROM Constante WHERE Sociedad='PE01' AND Programa='PLANTILLA' AND
            ((Campo='BANCO' AND Correlativo=@IdBanco) OR (Campo='TIPO_TRANSFERENCIA' AND Correlativo=@IdTransferencia) OR (Campo='TIPO_MONEDA' AND Correlativo=@IdMoneda2))
            """, request, tx, cancellationToken: ct));
        if (ejecutorValido == 0 || catalogosValidos != 3) throw new ArgumentException("Seleccione ejecutor, banco, medio de pago y moneda válidos.");
        var medio = await cn.QuerySingleAsync<string>(new CommandDefinition("SELECT ValorIni FROM Constante WHERE Sociedad='PE01' AND Programa='PLANTILLA' AND Campo='TIPO_TRANSFERENCIA' AND Correlativo=@IdTransferencia", request, tx, cancellationToken: ct));
        if (medio.Contains("CHEQUE", StringComparison.OrdinalIgnoreCase) ? string.IsNullOrWhiteSpace(request.Cheque) : !medio.Contains("EFECTIVO", StringComparison.OrdinalIgnoreCase) && string.IsNullOrWhiteSpace(request.NroOperacion))
            throw new ArgumentException("Ingrese el número de cheque o de operación según el medio de pago.");

        var items = new DataTable();
        items.Columns.Add("Correlativo", typeof(int));
        items.Columns.Add("IdSite", typeof(string));
        foreach (var item in request.Items) items.Rows.Add(item.Correlativo, item.IdSite);
        var parameters = new DynamicParameters();
        parameters.Add("IdEjecutor", request.IdEjecutor);
        parameters.Add("IdTransferencia", request.IdTransferencia);
        parameters.Add("IdBanco", request.IdBanco);
        parameters.Add("IdMoneda2", request.IdMoneda2);
        parameters.Add("FechaDeposito", request.FechaDeposito.Date, DbType.Date);
        parameters.Add("Cheque", request.Cheque.Trim(), DbType.AnsiString, size: 100);
        parameters.Add("NroOperacion", request.NroOperacion.Trim(), DbType.AnsiString, size: 100);
        parameters.Add("Usuario", usuario, DbType.AnsiString, size: 100);
        parameters.Add("FechaCreacion", DateTime.UtcNow.AddHours(-5));
        parameters.Add("Items", items.AsTableValuedParameter("dbo.PlanillaRevisionType"));
        var sp = "dbo.sp_Planilla_AdministrativoMasivo";
        if (request.EstadoOrigen == 8)
        {
            sp = "dbo.sp_Planilla_PagoContabilidadMasivo";
            parameters.Add("Opc", 2);
            parameters.Add("ObsAdicional", request.Comentario.Trim(), DbType.AnsiString);
            parameters.Add("ComentarioLog", request.Comentario.Trim(), DbType.AnsiString);
        }
        else parameters.Add("ComentarioAdicional", request.Comentario.Trim(), DbType.AnsiString);
        await cn.ExecuteAsync(new CommandDefinition(sp, parameters, tx, commandTimeout: 180, commandType: CommandType.StoredProcedure, cancellationToken: ct));
        var pagados = await cn.ExecuteScalarAsync<int>(new CommandDefinition("SELECT COUNT(*) FROM Planilla WHERE Correlativo IN @ids AND Estado=4", new { ids }, tx, cancellationToken: ct));
        if (pagados != ids.Length) throw new InvalidOperationException("El procedimiento no confirmó todos los pagos. La operación se revirtió.");
        await tx.CommitAsync(ct);
        return pagados;
    }

    private sealed class PagoActual
    {
        public int Correlativo { get; set; }
        public string IdSite { get; set; } = "";
        public int Estado { get; set; }
        public int TipoMoneda { get; set; }
        public decimal TotalPagar { get; set; }
        public string Version { get; set; } = "";
    }
}

File: cjerp-frontend/src/api/pagoTesoreriaService.ts
import httpClient from "./httpClient";

export type PagoTesoreriaRow = {
  idAnticipo: number | null;
  correlativo: number;
  idSite: string;
  estado: number;
  tipoMoneda: number;
  idResponsable: number;
  fecha: string | null;
  fechaDeposito: string | null;
  ot: string | null;
  responsable: string | null;
  solicitante: string | null;
  cliente: string | null;
  proyecto: string | null;
  site: string | null;
  tipoTrabajo: string | null;
  tarea: string | null;
  comprobante: string | null;
  moneda: string | null;
  subtotal: number;
  igv: number;
  total: number;
  montoRetencion: number;
  totalPagar: number;
  detalle: string | null;
  serie: string | null;
  banco: string | null;
  bancoCta?: string | null;
  transferencia: string | null;
  nroOperacion: string | null;
  cheque: string | null;
  comentarioAdicional: string | null;
  version: string;
  idComprobante: number | null;
  idTipoPago: number | null;
  ruc: string | null;
  fecEmision: string | null;
  idRendicion: number | null;
  rendicion: string | null;
  idRetencion: number | null;
  idBanco: number | null;
  idBancoCta: number | null;
  cuenta: string | null;
  cuentaInter: string | null;
  nombreCta: string | null;
  idMoneda2: number | null;
  idTransferencia: number | null;
  revisionPm: string | null;
  revisionPmAprobar: string | null;
  fechaRevisionAprobar: string | null;
  fechaRevision: string | null;
  observacion: string | null;
  imgFactura: string | null;
};
export type PagoOpcion = {
  id: number;
  nombre: string;
  porcentaje?: number | null;
};
export type PagoCatalogos = {
  anticipos: PagoOpcion[];
  estados: PagoOpcion[];
  ejecutores: PagoOpcion[];
  bancos: PagoOpcion[];
  transferencias: PagoOpcion[];
  monedas: PagoOpcion[];
  retenciones: PagoOpcion[];
  comprobantes: PagoOpcion[];
  tiposPago: PagoOpcion[];
  rendiciones: PagoOpcion[];
};
export type PagoTesoreriaRequest = {
  estadoOrigen: number;
  idEjecutor: number;
  idTransferencia: number;
  idBanco: number;
  idMoneda2: number;
  fechaDeposito: string;
  cheque: string;
  nroOperacion: string;
  comentario: string;
  items: PagoAccionItem[];
};
const url = "/tesoreria/pagos";
export type PagoRevisionPermisos = { puedeEditar: boolean; puedeEditarOperacion: boolean; puedeEditarEstado: boolean };
export type PagoRevisionRequest = Pick<PagoTesoreriaRow, "idAnticipo" | "nroOperacion" | "idComprobante" | "idTipoPago" | "imgFactura" | "estado"> & {
  item: PagoAccionItem;
  confirmarCambioEstado: boolean;
};
export const guardarRevisionTesoreria = (request: PagoRevisionRequest) =>
  httpClient.put<{ procesados: number }>(`${url}/revision`, request);

export const descargarFacturaRevision = (correlativo: number) =>
  httpClient.get<Blob>(`${url}/revision/${correlativo}/factura`, { responseType: "blob" });

// Mismo endpoint y almacenamiento que la página Gastos.
export const subirFacturaRevision = (archivo: File, row: PagoTesoreriaRow) => {
  const data = new FormData();
  data.append("archivo", archivo);
  data.append("gastoId", String(row.correlativo));
  if (row.serie) data.append("serie", row.serie);
  if (row.responsable) data.append("responsable", row.responsable);
  return httpClient.post<{ fileUrl: string; storagePath: string }>("/tesoreria/gastos/upload-factura", data);
};

export const obtenerCatalogosPago = (signal?: AbortSignal) =>
  httpClient.get<{ catalogos: PagoCatalogos; puedePagar: boolean; permisosRevision: PagoRevisionPermisos }>(
    `${url}/catalogos`,
    { signal },
  );
export const listarPagosTesoreria = (
  estado: number,
  desde: string,
  hasta: string,
  signal?: AbortSignal,
  correlativo?: number,
) =>
  httpClient.get<PagoTesoreriaRow[]>(url, {
    params: { estado, desde: desde || undefined, hasta: hasta || undefined, correlativo: correlativo || undefined },
    signal,
    timeout: 90000,
  });
export const listarPagosTesoreriaV1 = (estado: number, _desde = "", _hasta = "", signal?: AbortSignal, correlativoFiltro?: number, idEstadoFiltro?: number) =>
  httpClient.get<Record<string, unknown>[]>(`${url}/v1`, { params: { idEstado: estado === 100 ? idEstadoFiltro : estado, fechaInicio: _desde || undefined, fechaFin: _hasta || undefined }, signal, timeout: 90000 }).then((rows) => (Array.isArray(rows) ? rows : []).map((raw) => { const r = new Proxy(raw, { get: (target, prop: string) => target[prop] ?? target[Object.keys(target).find((key) => key.toLowerCase() === prop.toLowerCase()) ?? ""] }); return ({
    ...r,
    correlativo: Number(r.Corre ?? r.correlativo ?? 0), idSite: String(r.IdSite ?? r.idSite ?? ""), estado: Number(r.Estado ?? r.estado ?? 0), tipoMoneda: Number(r.TipoMoneda ?? r.tipoMoneda ?? 0), idResponsable: Number(r.IdResponsable ?? 0), idAnticipo: r.IdAnticipo == null ? null : Number(r.IdAnticipo), fecha: String(r.FecIngreso ?? ""), fechaDeposito: String(r.FechaDeposito ?? ""), ot: String(r.Ot ?? ""), responsable: String(r.Responsable ?? r.NomResponsable ?? ""), solicitante: String(r.Solicitante ?? ""), cliente: String(r.Cliente ?? ""), proyecto: String(r.NombreProyecto ?? ""), site: String(r.Site ?? ""), tipoTrabajo: String(r.Tipo_Trabajo ?? ""), tarea: String(r.Tarea ?? ""), comprobante: String(r.Comprobante ?? ""), moneda: String(r.Moneda ?? ""), subtotal: Number(r.Subtotal ?? 0), igv: Number(r.IGV ?? 0), total: Number(r.Total ?? 0), montoRetencion: Number(r.MontoRetencion ?? 0), totalPagar: Number(r.TotalPagar ?? 0), detalle: String(r.Detalle ?? ""), serie: String(r.Serie ?? ""), banco: String(r.Banco ?? ""), bancoCta: String(r.BancoCta ?? ""), transferencia: String(r.Transferencia ?? ""), nroOperacion: String(r.NroOperacion ?? ""), cheque: String(r.IdCheque ?? ""), comentarioAdicional: String(r.Comentario ?? ""), idComprobante: r.IdComprobante == null ? null : Number(r.IdComprobante), idTipoPago: r.IdTipoPago == null ? null : Number(r.IdTipoPago), ruc: String(r.RUC ?? ""), fecEmision: String(r.FecEmision ?? ""), idRendicion: r.IdRendicion == null ? null : Number(r.IdRendicion), rendicion: String(r.Rendicion ?? ""), idRetencion: r.IdRetencion == null ? null : Number(r.IdRetencion), idBanco: r.IdBanco == null ? null : Number(r.IdBanco), idBancoCta: r.IdBancoCta == null ? null : Number(r.IdBancoCta), cuenta: String(r.Cuenta ?? ""), cuentaInter: String(r.CuentaInter ?? ""), nombreCta: String(r.NombreCta ?? ""), idMoneda2: r.IdMoneda2 == null ? null : Number(r.IdMoneda2), idTransferencia: r.IdTransferencia == null ? null : Number(r.IdTransferencia), revisionPm: String(r.RevisionPm ?? ""), revisionPmAprobar: String(r.RevisionPmAprobar ?? ""), fechaRevisionAprobar: String(r.FechaReAprobador ?? ""), fechaRevision: String(r.FechaAprobador ?? ""), observacion: String(r.Observacion ?? ""), imgFactura: String(r.imgfactura ?? r.ImgFactura ?? ""), version: "v1",
  } as PagoTesoreriaRow); }).filter((r) => !correlativoFiltro || r.correlativo === correlativoFiltro));
export const obtenerCuentasPago = (
  idResponsable: number,
  signal?: AbortSignal,
) =>
  httpClient.get<
    {
      cuenta: string | null;
      cuentaInter: string | null;
      nombreCta: string | null;
      banco: string | null;
    }[]
  >(`${url}/cuentas/${idResponsable}`, { signal });
export const registrarPagosTesoreria = (request: PagoTesoreriaRequest) =>
  httpClient.post<{ procesados: number }>(url, request, { timeout: 210000 });
export const grabarPagoTesoreria = (request: Omit<PagoTesoreriaRequest, "estadoOrigen">) =>
  httpClient.post<{ procesados: number }>(`${url}/grabar`, request, { timeout: 60000 });

export type PagoAccionItem = {
  correlativo: number;
  idSite: string;
  estado: number;
  version: string;
  totalPagar: number;
};
export type PagoAccion =
  | "revisar"
  | "contabilidad-programar"
  | "contabilidad-administrativo"
  | "programar"
  | "administrativo"
  | "observar"
  | "rendicion"
  | "corregir"
  | "subsanar";
export type PagoAccionRequest = {
  accion: PagoAccion;
  items: PagoAccionItem[];
  observacion: string;
  idRetencion?: number;
  aplicarGenerales?: boolean;
  aplicarBanco?: boolean;
  aplicarRendicion?: boolean;
  aplicarOperacion?: boolean;
  idComprobante?: number;
  idTipoPago?: number;
  ruc?: string;
  serie?: string;
  fecEmision?: string;
  idBanco?: number;
  idMoneda2?: number;
  fechaDeposito?: string;
  idTransferencia?: number;
  nroOperacion?: string;
  idRendicion?: number;
  aplicarDetalle?: boolean;
  detalle?: string;
  aplicarAdjunto?: boolean;
  imgFactura?: string;
};
export const ejecutarAccionTesoreria = (request: PagoAccionRequest) =>
  httpClient.post<{ procesados: number }>(`${url}/acciones`, request, {
    timeout: 210000,
  });
export const crearItemsTesoreria = (
  rows: PagoTesoreriaRow[],
): PagoAccionItem[] =>
  rows.map((r) => ({
    correlativo: r.correlativo,
    idSite: r.idSite,
    estado: r.estado,
    version: r.version,
    totalPagar: r.totalPagar,
  }));
